### PR TITLE
[MPS] Add error inputs check

### PR DESCRIPTION
**Pull request by qqaatw**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: test/test_mps.py [hunk 1]
```diff
@@ -38,13 +38,14 @@
...
 )
-from torch.testing._internal.common_device_type import ops, dtypes, instantiate_device_type_tests
+from torch.testing._internal.common_device_type import ops, dtypes, instantiate_device_type_tests, OpDTypes
 from torch.testing._internal.common_nn import NNTestCase
 import numpy as np
 import torch
 import torch.utils._pytree as pytree
 from itertools import product
 
+test_consistency_op_db = copy.deepcopy(op_db)
```
Comment (qqaatw) [inline test/test_mps.py:48]: op_db for TestConsistency will be in-place modified, deep-copy one for it to prevent other tests from being affected.

File: test/test_mps.py [hunk 2]
```diff
@@ -10241,7 +10300,7 @@ def get_samples():
             self.assertEqual(cpu_out, mps_out, atol=atol, rtol=rtol)
 
 
-    @ops(mps_ops_grad_modifier(copy.deepcopy(op_db)), allowed_dtypes=MPS_GRAD_DTYPES)
+    @ops(mps_ops_grad_modifier(copy.deepcopy(test_consistency_op_db)), allowed_dtypes=MPS_GRAD_DTYPES)
```
Comment (kulinseth) [inline test/test_mps.py:10303]: We are already doing a deep_copy where it's needed, have you seen any issues? Why do we need to copy it again.

File: test/test_mps.py [hunk 3]
```diff
@@ -744,6 +746,63 @@ def addDecorator(op, d) -> None:
                          dtypes=MACOS_12_3_XFAILLIST[key]))
         yield op
 
+def mps_ops_error_inputs_modifier(ops):
+    SKIPLIST = {
```
Comment (kulinseth) [inline test/test_mps.py:750]: SKIPLIST is not the way the to go. We have an expected fail list which should is where unimplemented ops and others are captured.

**Review by kulinseth — CHANGES_REQUESTED** [review]

File: test/test_mps.py (hunk 3 above)
Comment (qqaatw) [reply test/test_mps.py:750]: These tests are skipped because the op crashes that cannot be xfailed.

There is a TODO comment below stating the situation.

File: test/test_mps.py (hunk 2 above)
Comment (qqaatw) [reply test/test_mps.py:10303]: I left a comment here: https://github.com/pytorch/pytorch/pull/98167#discussion_r1155285682

In short, the op_db will be modified by the modifier which will cause unexpected xfail to other tests.

Comment (qqaatw) [reply test/test_mps.py:10303]: test_output_match doesn't have op_db deep-copied.

File: test/test_mps.py (hunk 3 above)
Comment (malfet) [reply test/test_mps.py:750]: @qqaatw are there are issues describing the crashes? If so, can be referenced in the SKIPLIST comments?
Otherwise, it gives false sense of safety: no open issues in https://github.com/pytorch/pytorch/issues?q=is%3Aopen+is%3Aissue+label%3A%22module%3A+mps%22+label%3A%22module%3A+crash%22 and all tests are green

Comment (qqaatw) [reply test/test_mps.py:750]: I think currently no, they're newly tested (probably edge) cases. I plan to fix those myself after this pr gets merged and file issues if a crash is not solvable on PyTorch side.

Comment (qqaatw) [reply test/test_mps.py]: Added checks to prevent crashes and removed `SKIPLIST`.

**Review by kulinseth — COMMENTED** [review]
Modulo the comment fix it looks good.

**Review by kulinseth — APPROVED** [review]

File: aten/src/ATen/native/mps/operations/Indexing.mm [hunk 4]
```diff
@@ -622,6 +622,25 @@ Tensor index_select_mps(const Tensor& self, int64_t dim, const Tensor& index) {
   TORCH_CHECK(self.scalar_type() == output.scalar_type(),
               "index_select(): self and output must have the same scalar type");
   TORCH_CHECK(dim == 0 || dim < self.dim(), "index_select(): Indexing dim ", dim, " is out of bounds of tensor");
+  TORCH_CHECK(output.dim() == 0 || index.size(-1) == output.size(dim),
+              "index_select(): index and output must have the same size at `dim`th dimension, but got ",
+              index.size(-1),
+              " and ",
+              output.size(-1),
```
Comment (qqaatw) [inline aten/src/ATen/native/mps/operations/Indexing.mm:629]:
```suggestion
              output.size(dim),
```
